specs/lightclient/beacon-chain.md

```diff
@@ -0,0 +1,202 @@
+
+This is a standalone beacon chain patch adding light client support via sync committees.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `CompactValidator` | `uint64` | a compact validator |
+
+## Constants
+
+| Name | Value |
+| - | - | 
+| `BASE_REWARDS_PER_EPOCH` | `uint64(5)` |
+
+## Configuration
+
+### Misc
+
+| Name | Value |
+| - | - | 
+| `MAX_SYNC_COMMITTEE_SIZE` | `uint64(2**8)` (= 256) |
+
+### Time parameters
+
+| Name | Value | Unit | Duration |
+| - | - | :-: | :-: |
+| `EPOCHS_PER_SYNC_COMMITTEE_PERIOD` | `Epoch(2**8)` (= 256) | epochs | ~27 hours |
+
+### Domain types
+
+| Name | Value |
+| - | - |
+| `DOMAIN_SYNC_COMMITTEE` | `DomainType('0x07000000')` |
+
+## Containers
+
+### Extended containers
+
+#### `BeaconBlockBody`
+
+```python
+class BeaconBlockBody(phase0.BeaconBlockBody):
+    sync_committee_bits: Bitlist[MAX_SYNC_COMMITTEE_SIZE]
+    sync_committee_signature: BLSSignature
+```
+
+#### `BeaconState`
+
+```python
+class BeaconState(phase0.BeaconState):
+    current_sync_committee: SyncCommittee
+    next_sync_committee: SyncCommittee
+```
+
+### New containers
+
+#### `SyncCommittee`
+
+```python
+class SyncCommittee(Container):
+    pubkeys: List[BLSPubkey, MAX_SYNC_COMMITTEE_SIZE]
+    pubkeys_aggregate: BLSPubkey
+    compact_validators: List[CompactValidator, MAX_SYNC_COMMITTEE_SIZE]
+```
+
+## Helper functions
+
+### Misc
+
+#### `compactify_validator`
+
+```python
+def compactify_validator(index: ValidatorIndex, slashed: bool, effective_balance: Gwei) -> CompactValidator:
+    """
+    Return the compact validator for a given validator index, slashed status and effective balance.
+    """
+    return CompactValidator((index << 16) + (slashed << 15) + uint64(effective_balance // EFFECTIVE_BALANCE_INCREMENT))
+```
+
+#### `decompactify_validator`
+
+```python
+def decompactify_validator(compact_validator: CompactValidator) -> Tuple[ValidatorIndex, bool, Gwei]:
+    """
+    Return the validator index, slashed status and effective balance for a given compact validator.
+    """
+    index = ValidatorIndex(compact_validator >> 16)  # from bits 16-63
+    slashed = bool((compact_validator >> 15) % 2)  # from bit 15
+    effective_balance = Gwei(compact_validator & (2**15 - 1)) * EFFECTIVE_BALANCE_INCREMENT  # from bits 0-14
+    return (index, slashed, effective_balance)
+```
+
+### Beacon state accessors
+
+#### `get_sync_committee_indices`
+
+```python
+def get_sync_committee_indices(state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
+    """
+    Return the sync committee indices for a given state and epoch.
+    """
+    start_epoch = Epoch((max(epoch // EPOCHS_PER_SYNC_COMMITTEE_PERIOD, 1) - 1) * EPOCHS_PER_SYNC_COMMITTEE_PERIOD)
+    active_validator_count = uint64(len(get_active_validator_indices(state, start_epoch)))
+    sync_committee_size = min(active_validator_count, MAX_SYNC_COMMITTEE_SIZE)
+    seed = get_seed(state, base_epoch, DOMAIN_SYNC_COMMITTEE)
```

Comment:
`start_epoch`? Fixed in https://github.com/ethereum/eth2.0-specs/pull/2133
```suggestion
    seed = get_seed(state, start_epoch, DOMAIN_SYNC_COMMITTEE)
```

Reply:
I knew I must have introduced a bug somewhere 😂 

@hwwhww: Does everything else pass? For example, are all the types casted properly and the ToC properly built?

Reply:
Used `base_epoch` everywhere!